src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -371,6 +372,35 @@ public Response getDataset(@PathParam("datasetId") Integer datasetId) {
     }
   }
 
+  @GET
+  @Path("/study/{studyId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  public Response getStudyById(@PathParam("studyId") Integer studyId) {
+    try {
+      Study study = datasetService.findStudyById(studyId);
+      if (Objects.isNull(study)) {
+        throw new NotFoundException("Could not find the study with id: " + studyId.toString());
+      }
+
+      // if datasets exist, get full datasets and add to study
+      Set<Integer> datasetIds = study.getDatasetIds();
+
+      if(Objects.isNull(datasetIds)){
+        throw new NotFoundException("Study has no datasets.");
+      }
+      List<Integer> datasetIdList = new ArrayList<>(datasetIds);
+      List<Dataset> datasetSet = datasetService.findDatasetsByIds(datasetIdList);
```

Comment:
In theory, there should always be dataset ids, but in practice, we often have incomplete data in dev. The underlying call passes that list to `datasetDAO.findDatasetsByIdList(datasetIds)` and if it's empty, you'll get a sql error. We should handle that by checking if the list is empty and not making an erroneous call in that edge case.